data/provider/src/main/java/tech/pegasys/teku/api/ObjectAndMetaData.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.api;
+
+import com.google.common.base.MoreObjects;
+import java.util.Objects;
+import tech.pegasys.teku.spec.SpecMilestone;
+
+public class ObjectAndMetaData<T> {
```

Comment:
This class is more generic than we need right now but the next step is to add beacon state with metadata and we don't want to have two separate definitions of the metadata so introducing a class that can be shared now. Little bit of a YAGNI violation but I'm feeling lucky. :)